core/src/test/java/org/elasticsearch/plugins/PluginManagerUnitTests.java

```diff
@@ -123,10 +123,9 @@ public void testTrimmingElasticsearchFromOfficialPluginName() throws IOException
     @Test
     public void testTrimmingElasticsearchFromGithubPluginName() throws IOException {
         String user = randomAsciiOfLength(6);
-        String randomName = randomAsciiOfLength(10);
-        String pluginName = randomFrom("elasticsearch-", "es-") + randomName;
+        String pluginName = randomAsciiOfLength(10);
```

Comment:
Iet's take `head` as an example.

> bin/plugin install mobz/elasticsearch-head

It was installed previously as `head` site plugin.

This plugin needs now to add a property file within its distribution telling that the plugin name is actually `head` even if the repo name is elasticsearch-head. This is the purpose of the PR you have created. Rely on property file and not anymore on the filename.

So you will still install it with:

> bin/plugin install mobz/elasticsearch-head

And it will still be `head` unless his author decides of another name like `es-head` or whatever...

Makes sense?